lib/functions.py

```diff
@@ -417,41 +390,31 @@ def filter_pattern(doc_identifier):
             return 'numbers'
     return None
 
-def filter_chapter(doc, language, session):
-    if session['cancellation_requested']:
-        stop_and_detach_tts()
-        raise ValueError('Cancel requested')
-
+def filter_chapter(doc, language):
     soup = BeautifulSoup(doc.get_body_content(), 'html.parser')
     # Remove scripts and styles
     for script in soup(["script", "style"]):
         script.decompose()
-
     # Normalize lines and remove unnecessary spaces
     text = re.sub(r'(\r\n|\r|\n){3,}', '\r\n', soup.get_text().strip())
     text = replace_roman_numbers(text)
     lines = (line.strip() for line in text.splitlines())
     chunks = (phrase.strip() for line in lines for phrase in line.split("  "))
     text = '\n'.join(chunk for chunk in chunks if chunk)
     text = text.replace('»', '"').replace('«', '"')
-
     # Pattern 1: Add a space between UTF-8 characters and numbers
     text = re.sub(r'(?<=[\p{L}])(?=\d)|(?<=\d)(?=[\p{L}])', ' ', text)
-
     # Pattern 2: Split numbers into groups of 4
     text = re.sub(r'(\d{4})(?=\d)', r'\1 ', text)
-
-    chapter_sentences = get_sentences(text, language, session)
+    chapter_sentences = get_sentences(text, language)
     return chapter_sentences
 
-def get_sentences(sentence, language, session, max_pauses=9):
+def get_sentences(sentence, language, max_pauses=9):
     max_length = language_mapping[language]['char_limit']
     punctuation = language_mapping[language]['punctuation']
+    sentence = sentence.replace(".", ";\n")
```

Comment:
This change broke sentence splitting for me. I also don't think it makes sense to replace all periods with semicolons...

Reply:
this is done on purpose to avoid a coqui-tts bug on several languages that coqui-tts dev didn't solve for now

Reply:
I think the biggest issue is that semicolons aren't prioritized like periods were. They're grouped with general punctuation. So instead of splitting mainly by sentence, sentences are split randomly on any punctuation. This can result in short sentence fragments that the TTS engine struggles with. I found it repeating words or adding gibberish.

Semicolons and newlines also work produce poor results for abbreviations like "5 a.m." or "F.B.I.".

Reply:
it can be problematic indeed. I must find another way to bypass the coqui-tts bug.  it's very hard to have all perfect TTS in one shot by the way as each language has it's own punctuation style, + acronysm etc.... for now the dot char is pronounced by the half of xtts-v2 languages... which cannot stay like that.